tests/integration-tests/build.gradle.kts

```
@@ -65,9 +65,8 @@ fun configureApollo(generateKotlinModels: Boolean) {
               }
               "normalizer" -> {
                 generateFragmentImplementations.set(true)
-                customScalarsMapping.set(mapOf(
-                    "Date" to "kotlinx.datetime.LocalDate"
-                ))
+                mapScalar("Date", "kotlinx.datetime.LocalDate")
+                mapScalar("Instant", "kotlinx.datetime.Instant", "com.example.ScalarAdapters.InstantAdapter")
```

Could you save a wrapper here? 

```suggestion
                mapScalar("Instant", "kotlinx.datetime.Instant", "com.apollographql.apollo3.adapter.KotlinxInstantAdapter")
```

---

That's what I tried first, but this doesn't work in Java (needs `.INSTANCE`).  I thought of trying `@JvmStatic` on the funs but the compiler doesn't like it because they are `override` :)

---

I see. This is a failure at observing our [guidelines for Java interop](https://github.com/apollographql/apollo-kotlin/blob/a1536a269ffe5896f1b5290e55bd361e2d398861/CONTRIBUTING.md#L61). 

You could do something like this instead:

```kotlin
if (generateKotlinModels) {
    mapScalar("Instant", "kotlinx.datetime.Instant", "com.apollographql.apollo3.adapter.KotlinxInstantAdapter")
} else {
    mapScalar("Instant", "kotlinx.datetime.Instant", "com.apollographql.apollo3.adapter.KotlinxInstantAdapter.INSTANCE")
}
```

---

> This is a failure at observing our guidelines for Java interop.

Yeah 😿! To be fair, here a top level variable wouldn't helpe either (Kotlin: `com.apollographql.apollo3.adapter.KotlinxInstantAdapter` / Java: `com.apollographql.apollo3.adapter.KotlinTimeAdapters.KotlinxInstantAdapter`).

---

Right ! oh well life is hard 😅